src/main/java/htsjdk/samtools/cram/structure/Container.java

```diff
@@ -188,28 +196,22 @@ public void distributeIndexingParametersToSlices() {
             throw new CRAMException(String.format(format, landmarks.length, slices.length));
         }
 
-        if (containerByteSize == 0) {
-            throw new CRAMException("Cannot set Slice indexing parameters if this Container's byte size is unknown");
+        if (containerBlocksByteSize == 0) {
+            throw new CRAMException("Cannot set Slice indexing parameters if the byte size of this Container's blocks is unknown");
         }
 
         final int lastSliceIndex = slices.length - 1;
         for (int i = 0; i < lastSliceIndex; i++) {
             final Slice slice = slices[i];
             slice.index = i;
-            slice.byteOffsetFromContainer = landmarks[i];
-            slice.byteSize = landmarks[i + 1] - slice.byteOffsetFromContainer;
+            slice.byteOffsetFromCompressionHeaderStart = landmarks[i];
+            slice.byteSize = landmarks[i + 1] - slice.byteOffsetFromCompressionHeaderStart;
         }
 
         final Slice lastSlice = slices[lastSliceIndex];
         lastSlice.index = lastSliceIndex;
-        lastSlice.byteOffsetFromContainer = landmarks[lastSliceIndex];
-
-        // calculate a "final landmark" indicating the byte offset of the end of the container
```

Comment:
no.  `containerBlocksByteSize` is already equal to that "final landmark".

This is the line that reverts #1326